sdk/lusid/api_client.py

```diff
@@ -610,12 +622,17 @@ def __deserialize_primitive(self, data, klass):
 
         :return: int, long, float, str, bool.
         """
-        try:
-            return klass(data)
-        except UnicodeEncodeError:
-            return six.text_type(data)
-        except TypeError:
-            return data
+
+
+        if type(klass) is float and self.flag_to_use_decimal and len(get_floating_part(data)) > 15:
```

Comment:
Are there any problems that a user might encounter when they end up with a mix of floats and Decimals being used because of varying precision across properties?